### PR TITLE
(GH-1294) Use the puppet_agent module instead of a built in plugin.

### PR DESCRIPTION
**Pull request by adreyer**

### CODE REVIEW

**Comment by lucywyman** [comment]
@adreyer Looks to be missing a `task_plugin::apply_prep` fixture?

**Comment by adreyer** [comment]
yeah I'm just refactoring the task_spec tests so they look like the module_spec ones and don't need any special files.

**Comment by adreyer** [comment]
Why don't we install the puppetfile in the repo?

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,12 @@
 
 ## Bolt Next
 
+### Breaking Changes
```
Comment (K8med) [inline CHANGELOG.md]: Note: I haven't seen a "Breaking changes" header before, I'm checking that it's standard? Will we continue to use it as needed?

Comment (adreyer) [reply CHANGELOG.md]: hmm I think deprecations may be the standard header

**Pull request merged**